### ISSUE TITLE
Multiple suggestions for FGD autogen

### ISSUE THREAD

**Issue by AWildErin** [Facepunch/sbox-issues#357]
**What can't you do?**
So, with the new FGD gen we can't do things like 
`box_orientated(min, max, [auto center bool])`
`sphere(value)`
`wirebox(min, max)`
`line(color, start, end)`
`color(0, 0, 0)`
`size(min, max)` I think it's min max anyway.

These are some other things that are in the FGD that might just be useless, but either way I'll list them
`lightortho(range, width, height)`
`leansphere(radius, color)`
`base(base1, base2)` This might be useless, I haven't played much with the FGD stuff to know if bases would be required, but I'm listing it anyway

I could see these functions being good for visualising things in Hammer as we do with the lights and some other things. Since there's no actual code for Source 2 out yet, I'm not sure of all the stuff we can do with FGD, but these I've gathered by skimming base.fgd. I'm sure they'll probably help some people, or might entirely be useless, but I think they would be a neat addition people can use at least.

**How would you like it to work?**
This part I'm not entirely sure about, I made my own solution for this which worked as a way to do most of them, using like `[Hammer.Value("wirebox", 5, 5)]` though this only works for ones without commas so it would not work for line and color for example. 

So I think these might require some discussion on how it can be implemented and not be confusing for users.

**What have you tried?**
I've tried doing it manually in the FGD, but it gets override as usual. As far as I am aware it's not possible to do any of these without manually editing the FGD which gets autogen anyway so it's pointless to edit it.

If this is entirely useless for most people feel free to close it, I'm just listing off some features that *might* be useful to some users.

**Comment by robotboy655** [comment]
I think these should be added as needed. You can implement this kinda stuff yourself too:
```
namespace Hammer
{
	[AttributeUsage( AttributeTargets.Class )]
	public class MyHammerAttribute : MetaDataAttribute
	{
		internal string someData;

		public MyHammerAttribute( string someData )
		{
			this.someData = someData;
		}

		public override void AddHeader( StringBuilder sb )
		{
			// This is where the magic happens
			sb.Append( $"lightortho( {someData} ) " );
		}
	}
}
```

That being said I added some generic stuff like Hammer.Sphere and Hammer.Text. I will look into line/box later too.

I couldn't get  `color()` to do anything at all (nor can I find it in the source code, so I am assuming it doesn't exist), and `wirebox` is also janky in my tests so far.

**Comment by AWildErin** [comment]
Yeah, I have implemented a few myself. As for color and wirebox I just took what I saw inside the FGD. Color is used on a few entities in base.fgd (env_fire, sky_camera etc), and if there's no then it's probably unused. Unsure of what it'd be used for anyway, but some of the more generic ones should be implemented as people can get some use out of them.

I've implemented `box_world_oriented`, `box_world_aligned` and `size` as they all follow the same min, max parameters.

**Comment by robotboy655** [comment]
Ok, so, I added the following:

* `box_orientated` and its other variants as `[Hammer.BoundsHelper]`
* `sphere` and `leansphere` as `[Hammer.Sphere]`
* `line` and `selected_line` as `[Hammer.Line]`
* `size` as `[Hammer.BoxSize]`

I did not add the following because:

* `wirebox` as it seems like a broken/worse version of `box_orientated` and its variations.
* `color` as it only affects the classname text in 2D view which I find pointless. Let me know if you think otherwise.
* `base` as it serves no use in context of automatic .fgd generation, if anything its use would not be manual.

I am still not sure what to do about `lightortho`, it's only used for light entities at the moment, it might be useful for modders, but the name is really not helpful, maybe it needs renaming to something like `box_ortho`, or do we even care about the name as long as it's the attribute is appropriately in C#?
